src/udiskslinuxfilesystem.c

```diff
@@ -1886,12 +1924,21 @@ handle_unmount (UDisksFilesystem      *filesystem,
                  udisks_block_get_device (block),
                  caller_uid);
 
+  /* wait for mount-points update before returning from method */
+  filesystem_object = udisks_daemon_wait_for_object_sync (daemon,
```

Comment:
I don't like the idea of using `udisks_daemon_wait_for_object_sync` like this. It would be better to have a special "waiting" function for property changes. But I guess we can leave it this way for now and come with some better solution later when (and if) we decide to rewrite all functions to wait for the property change.